### DISCUSSION TITLE
How to remove outlier points in Ground Classification Points

### DISCUSSION THREAD

**Discussion by soadD** [r-lidar/lidR#753] [Q&A]
Struggling with some outliers in my PMF Ground Classification, where i can't filter out some  points. I've succeced in removing them when I've put the ground in another variable, with the funcion filter_noise, but had no luck with the las itself.

Any ideas with what to do ?
![errorcode](https://github.com/r-lidar/lidR/assets/109705634/dfaa6d49-993b-46e4-86ea-73666cb391e4)

My las keeps this way
![erro2](https://github.com/r-lidar/lidR/assets/109705634/3aea92aa-fb85-4a83-95e8-f1d4a0e588c8)
![errodtm](https://github.com/r-lidar/lidR/assets/109705634/17b40c72-4967-4eb1-9cdc-2af45263dbcd)

As opposed to what i've achieved filtering the ground
![limpo](https://github.com/r-lidar/lidR/assets/109705634/6c8d5199-5208-4a5c-a0d5-8805e96d3695)

edit.: I've changed some parameters in the pmf algorithm and but despite the new ground classification i got a good result when I made the dtm something weird happened, pics below:
new code
![novoerro1](https://github.com/r-lidar/lidR/assets/109705634/f89df697-34c9-4e00-acd8-d1d625d029fc)

Triangular irregular network
![novoerro2](https://github.com/r-lidar/lidR/assets/109705634/07206ec4-ae02-49cc-b6f1-d88262af7cf4)

Kriging
![krigingerror](https://github.com/r-lidar/lidR/assets/109705634/b9d0da40-bd67-4bd2-af41-1a60ae763387)

So, my question would be if i can post process the first ground classification as it has a better dtm representation besides the outliers.

**Comment by Jean-Romain** [comment]
With better parameters. `pmf` stands for Progressive Morphological Filter. Your inputs are not progressive: `pmf(1, 0.1)`.  At least you could just copy paste the documentation. At best you could read the original paper to understand what you are manipulating. Using  `pmf(1, 0.1)` you defeated the concept of the algorithm

**Reply by soadD** [reply]
I think your advice was little helpful, actually not helpful at all, sincerely i don't know why u responded in the first place, maybe cause u need to. Anyway, can't see why my parameters are bad, since i've got a fixed window of 1m² and lowering the treshold I was getting good results seeing crossections of it:

Before:
![first](https://github.com/r-lidar/lidR/assets/109705634/5cfda6c2-02b0-4695-a24c-b44cd0b13f9e)

After:
![second](https://github.com/r-lidar/lidR/assets/109705634/7eca97da-195c-4dd9-9749-d178e3d46f5e)

I got less points on bushes when i lowered the treshold and nothing seemed to bug until then. So my question still seems to be unresponded...

**Comment by bi0m3trics** [comment]
@soadD Have you actually tried using pmf with vectors (e.g., sequences) to see if it helps with your "noise" issue? Specifically, the help provides the following example for specifying the ws and th parameters, in which it specifies that both should be sequences:
```r
LASfile <- system.file("extdata", "Topography.laz", package="lidR")
las <- readLAS(LASfile, select = "xyzrn", filter = "-inside 273450 5274350 273550 5274450")

ws <- seq(3,12, 3)
th <- seq(0.1, 1.5, length.out = length(ws))

las <- classify_ground(las, pmf(ws, th))
```

Sure the book ([lidR book](https://r-lidar.github.io/lidRbook/gnd.html#pmf)) provides example where the parameters are both single values and sequences, but what JR was trying to say if that if you a single numerical value for each window and threshold parameter, it's not actually "progressive", it's just a "morphological filter" at that point and not really being used as intended.
